serde/src/java/org/apache/hadoop/hive/serde2/objectinspector/primitive/WritableConstantIntObjectInspector.java

```diff
@@ -46,12 +46,4 @@ public IntWritable getWritableConstantValue() {
     return value;
   }
 
-  @Override
-  public int precision() {
```

Comment:
I think this could lead to a change of behavior, e.g., all integers will have same precision by default, but constant integer would have the precision of the constant value?

Reply:
Yes, by overriding precision constant values has the precision of the value itself.